frame/contracts/src/migration.rs

```diff
@@ -527,20 +511,26 @@ mod test {
 
 	#[test]
 	fn is_upgrade_supported_works() {
-		type M = (MockMigration<9>, MockMigration<10>, MockMigration<11>);
+		type Migrations = (MockMigration<9>, MockMigration<10>, MockMigration<11>);
 
-		[(1, 1), (8, 11), (9, 11)].into_iter().for_each(|(from, to)| {
+		[(1, 1), (8, 11)].into_iter().for_each(|(from, to)| {
 			assert!(
-				M::is_upgrade_supported(StorageVersion::new(from), StorageVersion::new(to)),
+				Migrations::is_upgrade_supported(
+					StorageVersion::new(from),
+					StorageVersion::new(to)
+				),
 				"{} -> {} is supported",
 				from,
 				to
 			)
 		});
 
-		[(1, 0), (0, 3), (7, 11), (8, 10)].into_iter().for_each(|(from, to)| {
+		[(1, 0), (0, 3), (7, 11), (8, 10), (9, 11)].into_iter().for_each(|(from, to)| {
```

Comment:
why `(8,10)` is not supported?

Reply:
The test defines 
```rust
type Migrations = (MockMigration<9>, MockMigration<10>, MockMigration<11>);
```
so  8 to 10 is not valid, it should include all steps all the way to the latest version (11)

Reply:
> it should include all steps all the way to the latest version (11)

well, the question then is that why we can't run it partially eg to `target - 1`? what if we have migration code ready but need to hold on it from being run for a while?

Reply:
> it should include all steps all the way to the latest version (11)

so could we re-phrase this as a rule?: 
```rust
assert_eq!(pallet_contracts::STORAGE_VERSION, Migrations::VERSION_RANGE.1)
```

Reply:
Asking bc have not found this in docs, and the function doc is misleading on this regard.

Reply:
> ```rust
> assert_eq!(pallet_contracts::STORAGE_VERSION, Migrations::VERSION_RANGE.1)
>```
@pgherveou mb it would make sense to add this check to `MigrationSequence::integrity_test()` ?

Reply:
let's move this conversation to its own PR